master/buildbot/scripts/base.py

```diff
@@ -42,6 +43,31 @@ def captureErrors(errors, msg):
         defer.returnValue(1)
 
 
+def checkPidFile(pidfile):
+    """ mostly comes from _twistd_unix.py which is not twisted public API :-/
+
+        except it returns an exception instead of exiting
+    """
+    if os.path.exists(pidfile):
+        try:
+            with open(pidfile) as f:
+                pid = int(f.read())
+        except ValueError:
+            raise ValueError('Pidfile %s contains non-numeric value' % pidfile)
+        try:
+            os.kill(pid, 0)
+        except OSError as why:
+            if why.errno == errno.ESRCH:
+                # The pid doesn't exist.
+                print('Removing stale pidfile %s' % pidfile)
+                os.remove(pidfile)
+            else:
+                raise OSError("Can't check status of PID %s from pidfile %s: %s" %
+                              (pid, pidfile, why[1]))
+        else:
+            raise AssertionError("'%s' exists - is this master still running?")
```

Comment:
I thought about it for a while, and reasoned we are actually asserting there is not pidfile here.

I was looking for a BusyError or something, but I couldn't find it. So after looking again, looks like RuntimeError is for the error where you couldn't find anything more appropriate, so fair enough